[angle] Linker failure when using Angle on macOS

**Describe the bug**
I am using a package which uses Angle. I've included it in my CMakeLists.txt as follows:
FIND_PACKAGE(unofficial-angle REQUIRED)
TARGET_LINK_LIBRARIES(myLib
  unofficial::angle::libEGL
  unofficial::angle::libGLESv2
)

This causes a link error as shown below. I do not get the link error when I revert the following patch:
commit 2c27d320ef40bb188b6893100e790bdc9094d2de
Author: ayeteadoe <ayeteadoe@gmail.com>
Date:   Tue Aug 12 15:03:54 2025 -0700
[angle] Update to chromium_7258, add metal support, fix Linux/Windows build (#46841)

**Environment**
- OS: MacOS Sequoia Version 15.5
- Compiler:
Apple clang version 17.0.0 (clang-1700.0.13.5)
Target: arm64-apple-darwin24.5.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install angle
2. Build package which uses angle on macOS
3. Linker error occurs


**Expected behavior**
No Linker error to occur.

**Failure logs**
When building, I get the following linker error:

```
Undefined symbols for architecture arm64:
  "_MTLCreateSystemDefaultDevice", referenced from:
      angle::IsMetalRendererAvailable()::$_0::operator()() const in ANGLE.a[26](apple_platform_utils.mm.o)
  "sh::RewriteDoWhile(sh::TCompiler*, sh::TIntermNode*, sh::TSymbolTable*)", referenced from:
      sh::TCompiler::checkAndSimplifyAST(sh::TIntermBlock*, sh::TParseContext const&, ShCompileOptions const&) in ANGLE.a[61](Compiler.cpp.o)
  "sh::UnfoldShortCircuitAST(sh::TCompiler*, sh::TIntermBlock*)", referenced from:
      sh::TCompiler::checkAndSimplifyAST(sh::TIntermBlock*, sh::TParseContext const&, ShCompileOptions const&) in ANGLE.a[61](Compiler.cpp.o)
  "sh::RewriteRowMajorMatrices(sh::TCompiler*, sh::TIntermBlock*, sh::TSymbolTable*)", referenced from:
      sh::TranslatorGLSL::translate(sh::TIntermBlock*, ShCompileOptions const&, sh::PerformanceDiagnostics*) in ANGLE.a[51](TranslatorGLSL.cpp.o)
  "sh::AddAndTrueToLoopCondition(sh::TCompiler*, sh::TIntermNode*)", referenced from:
      sh::TCompiler::checkAndSimplifyAST(sh::TIntermBlock*, sh::TParseContext const&, ShCompileOptions const&) in ANGLE.a[61](Compiler.cpp.o)
  "sh::RewriteUnaryMinusOperatorFloat(sh::TCompiler*, sh::TIntermNode*)", referenced from:
      sh::TranslatorGLSL::translate(sh::TIntermBlock*, ShCompileOptions const&, sh::PerformanceDiagnostics*) in ANGLE.a[51](TranslatorGLSL.cpp.o)
ld: symbol(s) not found for architecture arm64
```

**Additional context**
N/A

--- a/ports/angle/cmake-buildsystem/PlatformMac.cmake
+++ b/ports/angle/cmake-buildsystem/PlatformMac.cmake
@@ -38,9 +38,15 @@ if(USE_OPENGL)
     list(APPEND ANGLE_SOURCES
         ${angle_translator_glsl_base_sources}
         ${angle_translator_glsl_sources}
+        ${angle_translator_glsl_apple_sources}
     )
     # Enable GLSL compiler output.
     list(APPEND ANGLE_DEFINITIONS ANGLE_ENABLE_GLSL ANGLE_ENABLE_GL_DESKTOP_BACKEND ANGLE_ENABLE_APPLE_WORKAROUNDS ANGLE_ENABLE_CGL)
+
+    # Still need to link with Metal as we call MTLCreateSystemDefaultDevice even if USE_METAL is not defined
+    list(APPEND ANGLEGLESv2_LIBRARIES
+        "-framework Metal"
+    )
 endif()
 
 if(USE_OPENGL OR ENABLE_WEBGL)

--- a/ports/angle/portfile.cmake
+++ b/ports/angle/portfile.cmake
@@ -32,7 +32,7 @@ endif()
 
 set(USE_METAL OFF)
 if ("metal" IN_LIST FEATURES)
-    set(USE_METAL ON)
+  set(USE_METAL ON)
 endif()
 
 # chromium/7258

--- a/ports/angle/vcpkg.json
+++ b/ports/angle/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "angle",
   "version-string": "chromium_7258",
-  "port-version": 1,
+  "port-version": 2,
   "description": [
     "A conformant OpenGL ES implementation for Windows, Mac and Linux.",
     "The goal of ANGLE is to allow users of multiple operating systems to seamlessly run WebGL and other OpenGL ES content by translating OpenGL ES API calls to one of the hardware-supported APIs available for that platform. ANGLE currently provides translation from OpenGL ES 2.0 and 3.0 to desktop OpenGL, OpenGL ES, Direct3D 9, and Direct3D 11. Support for translation from OpenGL ES to Vulkan is underway, and future plans include compute shader support (ES 3.1) and MacOS support."

--- a/versions/a-/angle.json
+++ b/versions/a-/angle.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "6eb27c13ec4328ed3d63f058a485a4e213087bda",
+      "version-string": "chromium_7258",
+      "port-version": 2
+    },
+    {
       "git-tree": "165d665818d070431c2889ac2b8f97f006948924",
       "version-string": "chromium_7258",
       "port-version": 1

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -138,7 +138,7 @@
     },
     "angle": {
       "baseline": "chromium_7258",
-      "port-version": 1
+      "port-version": 2
     },
     "ankurvdev-embedresource": {
       "baseline": "0.0.12",

